packages/react-composites/src/composites/ChatComposite/file-sharing/FileUpload.ts

```diff
@@ -0,0 +1,217 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+import { EventEmitter } from 'events';
+import { nanoid } from 'nanoid';
+
+/**
+ * Meta Data containing information about the uploaded file.
+ * @beta
+ */
+export interface FileMetadata {
+  /**
+   * File name to be displayed.
+   */
+  name: string;
+  /**
+   * Extension is used for rendering the file icon.
+   * An unknown extension will be rendered as a generic icon.
+   * Example: `jpeg`
+   */
+  extension: string;
+  /**
+   * Download URL for the file.
+   */
+  url: string;
+}
+
+/**
+ * @beta
+ */
+export interface FileUploadState {
+  /**
+   * Unique identifier for the file upload.
+   */
+  id: string;
+
+  /**
+   * Filename extracted from the {@link File} object.
+   */
+  filename: string;
```

Comment:
This duplicates `metadata.name`. If the intent is somehow different, please document how these two fields differ.

Reply:
Done